superset/views/core.py

```diff
@@ -2391,6 +2396,34 @@ def select_star(self, database_id, table_name, schema=None):
             mydb.select_star(table_name, schema, latest_partition=True, show_cols=True)
         )
 
+    @has_access_api
+    @expose("/estimate_query_cost/<database_id>/", methods=["POST"])
+    @expose("/estimate_query_cost/<database_id>/<schema>/", methods=["POST"])
+    @event_logger.log_this
+    def estimate_query_cost(self, database_id: int, schema: str = None) -> Response:
+        mydb = db.session.query(models.Database).filter_by(id=database_id).first()
```

Comment:
i think that `one_or_none()` is a bit clearer than `first()`, but it's just personal preference

Reply:
Ah, I wasn't familiar with `one_or_none()`, seems better suited. I'll change.